pep-0695.rst

```diff
@@ -875,6 +878,181 @@ incorrectly, because functions within a class do not normally have access to
 the enclosing class namespace. The implementation will retain a reference to
 the class namespace so that class-scoped names are resolved correctly.
 
+.. _695-scoping-behavior:
+
+Scoping Behavior
+----------------
+
+The new syntax requires a new kind of scope that behaves differently
+from existing scopes in Python. Thus, the new syntax cannot be described exactly in terms of
+existing Python scoping behavior. This section specifies these scopes
+further by reference to existing scoping behavior: the new scopes behave
+like function scopes, except for a number of minor differences listed below.
+
+All examples include functions introduced with the pseudo-keyword ``def695``.
+This keyword will not exist in the actual language; it is used to
+clarify that the new scopes are for the most part like function scopes.
+
+``def695`` scopes differ from regular function scopes in the following ways:
+
+- If a ``def965`` scope is immediately within a class scope, or within another
+  ``def695`` scope that is immediately within a class scope, then names defined
+  in that class scope can be accessed within the ``def695`` scope. (Regular functions,
+  by contranst, cannot access names defined within an enclosing class scope.)
+- The following constructs are disallowed anywhere within a ``def695`` scope, though
```

Comment:
```suggestion
- The following constructs are disallowed anywhere at the top level within a ``def695`` scope, though
```

Reply:
How about this instead:
```suggestion
- The following constructs are disallowed directly within a ``def695`` scope, though
```

Reply:
That also works, though I find it slightly less clear. Could also do this:

```suggestion
- The following constructs are disallowed at the top level in a ``def695`` scope, though
```

Reply:
I don't like using the term "top level" here because to me it strongly suggests the module scope.

Reply:
"outermost level"?

Reply:
That would sound like there are multiple levels of def695 scopes, and you can use a walrus in the inner level of def695 scopes. I think the "directly within" wording is fine, as the same sentence makes clear that the restriction doesn't apply in nested scopes of other kinds.